-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-47604/44109

ERROR: Build errors found during clang-tidy run.
```
src/PhysicsTools/HepMCCandAlgos/interface/MCTruthHelper.h:662:39: error: binding reference of type 'shared_ptr<...>' to value of type 'const shared_ptr<...>' drops 'const' qualifier [clang-diagnostic-error]
  662 |     for (HepMC3::ConstGenParticlePtr &mother : (p.production_vertex())->particles_in()) {
      |                                       ^      ~
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02881/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/stl_vector.h:878:7: note: selected 'begin' function with iterator type 'const_iterator' (aka '__normal_iterator<const std::shared_ptr<const HepMC3::GenParticle> *, std::vector<std::shared_ptr<const HepMC3::GenParticle>, std::allocator<std::shared_ptr<const HepMC3::GenParticle>>>>')
--
src/PhysicsTools/HepMCCandAlgos/interface/MCTruthHelper.h:705:39: error: binding reference of type 'shared_ptr<...>' to value of type 'const shared_ptr<...>' drops 'const' qualifier [clang-diagnostic-error]
  705 |     for (HepMC3::ConstGenParticlePtr &daughter : (p.end_vertex())->particles_out()) {
      |                                       ^        ~
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02881/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/stl_vector.h:878:7: note: selected 'begin' function with iterator type 'const_iterator' (aka '__normal_iterator<const std::shared_ptr<const HepMC3::GenParticle> *, std::vector<std::shared_ptr<const HepMC3::GenParticle>, std::allocator<std::shared_ptr<const HepMC3::GenParticle>>>>')
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:129: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/PhysicsTools/HepMCCandAlgos/interface/MCTruthHelper.h
+++ b/PhysicsTools/HepMCCandAlgos/interface/MCTruthHelper.h
@@ -659,7 +659,7 @@ template <typename P>
 const HepMC3::GenParticle *MCTruthHelper<P>::mother(const HepMC3::GenParticle &p, unsigned int imoth) const {
   if (numberOfMothers(p) > 0) {
     unsigned int imothi = 0;
-    for (HepMC3::ConstGenParticlePtr &mother : (p.production_vertex())->particles_in()) {
+    for (HepMC3::ConstGenParticlePtr& mother : (p.production_vertex())->particles_in()) {
       if(imothi == imoth) return mother.get();
       imothi++;
     }
@@ -702,7 +702,7 @@ template <typename P>
 const HepMC3::GenParticle *MCTruthHelper<P>::daughter(const HepMC3::GenParticle &p, unsigned int idau) const {
   if (numberOfDaughters(p) > 0) {
     unsigned int idaui = 0;
-    for (HepMC3::ConstGenParticlePtr &daughter : (p.end_vertex())->particles_out()) {
+    for (HepMC3::ConstGenParticlePtr& daughter : (p.end_vertex())->particles_out()) {
       if(idaui == idau) return daughter.get();
       idaui++;
     }

--- a/PhysicsTools/HepMCCandAlgos/interface/MCTruthHelper.h
+++ b/PhysicsTools/HepMCCandAlgos/interface/MCTruthHelper.h
@@ -659,7 +659,7 @@ template <typename P>
 const HepMC3::GenParticle *MCTruthHelper<P>::mother(const HepMC3::GenParticle &p, unsigned int imoth) const {
   if (numberOfMothers(p) > 0) {
     unsigned int imothi = 0;
-    for (HepMC3::ConstGenParticlePtr& mother : (p.production_vertex())->particles_in()) {
+    for (HepMC3::ConstGenParticlePtr mother : (p.production_vertex())->particles_in()) {
       if(imothi == imoth) return mother.get();
       imothi++;
     }
@@ -702,7 +702,7 @@ template <typename P>
 const HepMC3::GenParticle *MCTruthHelper<P>::daughter(const HepMC3::GenParticle &p, unsigned int idau) const {
   if (numberOfDaughters(p) > 0) {
     unsigned int idaui = 0;
-    for (HepMC3::ConstGenParticlePtr& daughter : (p.end_vertex())->particles_out()) {
+    for (HepMC3::ConstGenParticlePtr daughter : (p.end_vertex())->particles_out()) {
       if(idaui == idau) return daughter.get();
       idaui++;
     }

--- a/PhysicsTools/HepMCCandAlgos/interface/MCTruthHelper.h
+++ b/PhysicsTools/HepMCCandAlgos/interface/MCTruthHelper.h
@@ -659,8 +659,8 @@ template <typename P>
 const HepMC3::GenParticle *MCTruthHelper<P>::mother(const HepMC3::GenParticle &p, unsigned int imoth) const {
   if (numberOfMothers(p) > 0) {
     unsigned int imothi = 0;
-    for (HepMC3::ConstGenParticlePtr mother : (p.production_vertex())->particles_in()) {
-      if(imothi == imoth) return mother.get();
+    for (const HepMC3::ConstGenParticlePtr& mother : (p.production_vertex())->particles_in()) {
+      if (imothi == imoth) return mother.get();
       imothi++;
     }
   }
@@ -702,8 +702,8 @@ template <typename P>
 const HepMC3::GenParticle *MCTruthHelper<P>::daughter(const HepMC3::GenParticle &p, unsigned int idau) const {
   if (numberOfDaughters(p) > 0) {
     unsigned int idaui = 0;
-    for (HepMC3::ConstGenParticlePtr daughter : (p.end_vertex())->particles_out()) {
-      if(idaui == idau) return daughter.get();
+    for (const HepMC3::ConstGenParticlePtr& daughter : (p.end_vertex())->particles_out()) {
+      if (idaui == idau) return daughter.get();
       idaui++;
     }
   }

--- a/PhysicsTools/HepMCCandAlgos/plugins/GenParticleProducer.cc
+++ b/PhysicsTools/HepMCCandAlgos/plugins/GenParticleProducer.cc
@@ -355,7 +355,7 @@ void GenParticleProducer::produce(StreamID, Event& evt, const EventSetup& es) co
 
     if (ivhepmc == 3) {
       if (totalSize) {
-        for (HepMC3::GenVertexPtr v : mc3->vertices()) {
+        for (const HepMC3::GenVertexPtr& v : mc3->vertices()) {
           xyz0Ptr->SetXYZ((v->position()).x() * mmToCm, (v->position()).y() * mmToCm, (v->position()).z() * mmToCm);
           *t0Ptr = (v->position()).t() * mmToNs;
           break;
@@ -466,7 +466,7 @@ bool GenParticleProducer::fillDaughters3(reco::GenParticleCollection& cands,
                                          std::unordered_map<int, size_t>& barcodes) const {
   size_t numberOfMothers = (part->production_vertex())->particles_in_size();
   if (numberOfMothers > 0) {
-    for (HepMC3::ConstGenParticlePtr mother : (part->production_vertex())->particles_in()) {
+    for (const HepMC3::ConstGenParticlePtr& mother : (part->production_vertex())->particles_in()) {
       size_t m = barcodes.find(mother->id())->second;
       cands[m].addDaughter(GenParticleRef(ref, index));
       cands[index].addMother(GenParticleRef(ref, m));

--- a/PhysicsTools/HepMCCandAlgos/plugins/GenParticleProducer.cc
+++ b/PhysicsTools/HepMCCandAlgos/plugins/GenParticleProducer.cc
@@ -355,7 +355,7 @@ void GenParticleProducer::produce(StreamID, Event& evt, const EventSetup& es) co
 
     if (ivhepmc == 3) {
       if (totalSize) {
-        for (HepMC3::GenVertexPtr v : mc3->vertices()) {
+        for (HepMC3::GenVertexPtr& v : mc3->vertices()) {
           xyz0Ptr->SetXYZ((v->position()).x() * mmToCm, (v->position()).y() * mmToCm, (v->position()).z() * mmToCm);
           *t0Ptr = (v->position()).t() * mmToNs;
           break;
@@ -466,7 +466,7 @@ bool GenParticleProducer::fillDaughters3(reco::GenParticleCollection& cands,
                                          std::unordered_map<int, size_t>& barcodes) const {
   size_t numberOfMothers = (part->production_vertex())->particles_in_size();
   if (numberOfMothers > 0) {
-    for (HepMC3::ConstGenParticlePtr mother : (part->production_vertex())->particles_in()) {
+    for (HepMC3::ConstGenParticlePtr& mother : (part->production_vertex())->particles_in()) {
       size_t m = barcodes.find(mother->id())->second;
       cands[m].addDaughter(GenParticleRef(ref, index));
       cands[index].addMother(GenParticleRef(ref, m));

--- a/PhysicsTools/HepMCCandAlgos/plugins/GenParticleProducer.cc
+++ b/PhysicsTools/HepMCCandAlgos/plugins/GenParticleProducer.cc
@@ -355,7 +355,7 @@ void GenParticleProducer::produce(StreamID, Event& evt, const EventSetup& es) co
 
     if (ivhepmc == 3) {
       if (totalSize) {
-        for (HepMC3::GenVertexPtr& v : mc3->vertices()) {
+        for (HepMC3::GenVertexPtr v : mc3->vertices()) {
           xyz0Ptr->SetXYZ((v->position()).x() * mmToCm, (v->position()).y() * mmToCm, (v->position()).z() * mmToCm);
           *t0Ptr = (v->position()).t() * mmToNs;
           break;
@@ -466,7 +466,7 @@ bool GenParticleProducer::fillDaughters3(reco::GenParticleCollection& cands,
                                          std::unordered_map<int, size_t>& barcodes) const {
   size_t numberOfMothers = (part->production_vertex())->particles_in_size();
   if (numberOfMothers > 0) {
-    for (HepMC3::ConstGenParticlePtr& mother : (part->production_vertex())->particles_in()) {
+    for (HepMC3::ConstGenParticlePtr mother : (part->production_vertex())->particles_in()) {
       size_t m = barcodes.find(mother->id())->second;
       cands[m].addDaughter(GenParticleRef(ref, index));
       cands[index].addMother(GenParticleRef(ref, m));

